main/ui/src/main/java/org/cryptomator/ui/unlock/UnlockWorkflow.java

```diff
@@ -133,8 +129,20 @@ private void handleSuccess() {
 				window.setScene(successScene.get());
 				window.show();
 			});
-			case REVEAL -> vaultService.reveal(vault);
-			case IGNORE -> {}
+			case REVEAL -> {
+				Platform.runLater(() -> {
+					window.setScene(unlockScene.get());
+					window.close();
+				});
```

Comment:
```suggestion
				Platform.runLater(window::close);
```

No need to set a scene before closing a window. Especially not a scene that is already set anyway :wink:

Same for `IGNORE`

Reply:
Thanks for your reply. I set the scene intentionally, because I wasn’t sure if a `window.close` alone might close the wrong window, e.g. the main window.

Reply:
Will change it as you suggested and do some more testing :)

Reply:
With this change it is also working well, just as you've forseen it.